src/components/AmountFilter.jsx

```diff
@@ -48,8 +66,11 @@ export default function AmountFilter() {
           <Grid item className={classes.filterStyle}>
             <InputLabel className={classes.labelStyle}>Min Amount</InputLabel>
             <AmountTextField
-              value={values.min}
-              onChange={(e) => handleChange('min', e.target.value)}
+              error={error}
+              value={amountFilterVals.min || ''}
+              onChange={(e) =>
+                setMinMax('min', parseInt(e.target.value, 10) || 0)
```

Comment:
There is bug in the amount filter text field, when you try changing a number in between 2 other, it instead puts the new number at the end. For example, changing the 3 in `5352` to an 8 results to `5528`. I think this might be causing it?